Kernel/Locking/Mutex.cpp

```diff
@@ -18,6 +18,8 @@ void Mutex::lock(Mode mode, [[maybe_unused]] LockLocation const& location)
     // NOTE: This may be called from an interrupt handler (not an IRQ handler)
     // and also from within critical sections!
     VERIFY(!Processor::current_in_irq());
+    if constexpr (LOCK_DEBUG)
```

Comment:
is this going to make `LOCK_DEBUG` useless, looking at the commit message it seems like this will now/always crash in ext2 code?